### ISSUE TITLE
Implement normalization in sph_coeff matrix

### ISSUE THREAD

**Issue by anttimc** [bfieldtools/bfieldtools#22]
Currently, compute_sphcoeffs_mesh computes the linear transformations so that the resulting coefficients correspond to the default normalization (integration of ylm over unit sphere). When designing coils, it migth be better to have the coefficients s.t. for exampl,e sum_i(beta_i^2) would correspond to the energy of the field in a R-ball. This is called the energy normalization in other functions. 

In other words, we could have the same normalization schemes as in basis_fields function.

As a sidenote, we could also consider renaming this function: mesh_coef_matrix or similar

**Comment by rzetter** [comment]
Both 'energy' normalization and renaming the function sounds like a good idea!

**Comment by rzetter** [comment]
Closed by release of 0.9.0
